addons/mail/static/src/components/thread_view/thread_view.xml

```diff
@@ -23,6 +23,16 @@
                         t-ref="messageList"
                     />
                 </t>
+                <t t-elif="threadView.threadCache.hasLoadingFailed">
+                    <div class="o_ThreadView_loadingFailed">
+                        <div class="o_ThreadView_loadingFailedText">
+                            An error occured while fetching messages.
+                        </div>
+                        <button class="btn btn-link" t-on-click="_onClickRetryLoadMessages">
+                            Click here to retry
+                        </button>
```

Comment:
If not done yet please validate functionally with AJU with a testing step (both the concept of having a retry button and also its style & text).